ui-v2/src/components/ui/utils/spacing-utils.ts

```diff
@@ -0,0 +1,451 @@
+import { cva } from "class-variance-authority";
+
+type Displays =
+	| "hidden"
+	| "block"
+	| "contents"
+	| "flex"
+	| "flow-root"
+	| "grid"
+	| "inline"
+	| "inline-block"
+	| "inline-flex"
+	| "inline-grid"
+	| "list-item";
+
+type FlexDirection = "column" | "column-reverse" | "row-reverse" | "row";
+
+type AlignContent =
+	| "center"
+	| "start"
+	| "end"
+	| "stretch"
+	| "space-between"
+	| "space-around"
+	| "space-evenly";
+type AlignItems = "center" | "start" | "end" | "stretch" | "baseline";
+
+type AlignSelf = "auto" | "center" | "start" | "end" | "stretch" | "baseline";
+
+type Spaces =
+	| 0
+	| 0.5
+	| 1
+	| 1.5
+	| 2
+	| 2.5
+	| 3
+	| 4
+	| 5
+	| 6
+	| 8
+	| 10
+	| 12
+	| 14
+	| 16
+	| 20
+	| 24
+	| 32
+	| 48
+	| 64;
+
+export type UtilityProps = Partial<{
```

Comment:
If we want to add these props to our components in the future, we can just do something like:

```
type Props = ButtonProps & UtilityProps
...
return <Button className={cn(buttonVariants({ variant, size, className }), spacingUtiltiesClasses(props))}  ... />